File: Java/백준/Bronze/2798. 블랙잭/블랙잭.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken()),
				M = Integer.parseInt(st.nextToken());
		st = new StringTokenizer(br.readLine());
		int[] card = new int[N];
		for(int i = 0; i < N; i ++)
			card[i] = Integer.parseInt(st.nextToken());
		int max = 0;
		for(int x = 0 ; x < N; x ++) {
			for(int y = 0; y < N; y ++) {
				if(x == y) continue;
				for(int z = 0; z < N; z ++) {
					if(z == x || z == y) continue;
					if(card[x] + card[y] + card[z] <= M
							&& card[x] + card[y] + card[z] > max) {
						max = card[x] + card[y] + card[z];
					}
				}
			}
		}
		System.out.println(max);
	}
}

File: Kotlin (JVM)/백준/Bronze/2798. 블랙잭/블랙잭.kt
import java.util.StringTokenizer

fun main(args: Array<String>){
    var NM: StringTokenizer = StringTokenizer(readLine())
    var cards: StringTokenizer = StringTokenizer(readLine())
    var N: Int = NM.nextToken().toInt()
    var M: Int = NM.nextToken().toInt()
    var array: IntArray = IntArray(N)
    for (i: Int in array.indices)
        array[i] = cards.nextToken().toInt()
    var max = 0
    for(i: Int in array){
        var array2: IntArray = array.clone().filter { it != i }.toIntArray()
        for(j: Int in array2){
            var array3: IntArray = array2.clone().filter { it != j }.toIntArray()
            for(n: Int in array3){
                if(i+j+n in max+1.. M){
                    max = i+j+n
                }
            }
        }
    }
    println(max)
}